### PR TITLE
[SPARK-136] Refactor "RatMama" test suite

### PR DESCRIPTION
**Pull request by Eearslya**
Refactor and simplify the ratmama test suite, bringing the test count down from 900+ to just 12. This approach also avoids much of the monkeypatching required, reducing the risk of a refactor rendering these tests inoperable.

### CODE REVIEW

File: tests/fixtures/mock_bot.py [hunk 1]
```diff
@@ -77,6 +78,17 @@ def __init__(self, *args, **kwargs):
...
+            "some_announcer": {
+                "nickname": "some_announcer",
+                "username": "Announcer",
+                "hostname": "bot.fuelrats.com",
+                "away": False,
+                "away_message": None,
+                "account": "some_announcer",
+                "identified": True,
+
+                "realname": "RatMama"
```
Comment (theunkn0wn1) [inline tests/fixtures/mock_bot.py:85]: [style] stray newline in data

File: tests/unit/test_ratmama.py [hunk 2]
```diff
@@ -12,245 +12,185 @@
...
+                                         "#unit_test",
+                                         "some_recruit",
+                                         "ratsignal Sol, PC, O2 OK")
+    monkeypatch.setattr(context, "reply", async_callable_fx)
+
+    await ratmama.handle_ratsignal(context)
+    await ratmama.handle_ratsignal(context)
+
+    assert async_callable_fx.was_called_with("You already sent a signal, please be patient "
+                                             "while a dispatch is underway.")
```
Comment (theunkn0wn1) [inline tests/unit/test_ratmama.py]: [error] while i recognize this wasn't added by you, can you please adjust this test and the underlying implementation as to match the mecha2 interface while we are here?
```
<unknown> drillsignal ki pc
<Drillsqueak[BOT]> unknown: You already sent a Signal! Please stand by, someone will help you soon!
```

**Review by theunkn0wn1 — COMMENTED** [review]
Overall its looking great! thanks for writing this.

Two small things i would like to see addressed prior to merging this 🙂

File: src/packages/ratmama/ratmama_parser.py [hunk 3]
```diff
@@ -152,8 +154,8 @@
 
     for rescue in ctx.bot.board.rescues.values():
         if rescue.irc_nickname.casefold() == ctx.user.nickname.casefold():
-            await ctx.reply(
-                "You already sent a signal, please be patient while a dispatch is underway.")
+            await ctx.reply(f"{ctx.user.nickname}: You already sent a Signal! Please stand by,"
```
Comment (theunkn0wn1) [inline src/packages/ratmama/ratmama_parser.py:157]: [error] *squint* doesn't `ctx.reply` already embed the nickname into the message stream?

**Review by theunkn0wn1 — CHANGES_REQUESTED** [review]

**Review by theunkn0wn1 — APPROVED** [review]
LGTM

**Review by shatteredbeam — APPROVED** [review]
You did a ton of work on improving this,  Looks great.

**Comment by theunkn0wn1** [comment]
Update and this will be merged

**Pull request merged**